src/map/campaign_system.cpp

```diff
@@ -0,0 +1,364 @@
+/*
+===========================================================================
+
+Copyright (c) 2010-2015 Darkstar Dev Teams
```

Comment:
I like how we still live in the past for copyright headers.

Reply:
I thought about updating them all, but then I decided not to.